src/zenodo_to_gbif.Rmd

```diff
@@ -0,0 +1,110 @@
+---
+title: "Prepare Movebank data for GBIF IPT upload (make Darwin Core)"
+author:
+- Peter Desmet
+- Sanne Govaert
+date: "`r Sys.Date()`"
+output:
+  html_document
+---
+
+This script prepares Movebank bird tracking data for upload to a [GBIF IPT](https://gbif.org/ipt). It is designed for datasets deposited on Zenodo.
+
+1. Look up the DOI
+
+```{r setup, include = FALSE}
+knitr::opts_chunk$set(echo = TRUE, warning = FALSE, message = FALSE)
+```
+
+Load libraries:
+
+```{r}
+library(magrittr)
+library(here)
+library(movepub) # devtools::install_github("inbo/movepub")
+library(EML)
+library(jsonlite)
+```
+
+Set user-defined values:
+
+```{r}
+project_id <- "O_WESTERSCHELDE"
+versioned_doi <- "https://doi.org/10.5281/zenodo.5879096"
+rights_holder <- "INBO"
+contact <- person(
+  given = "Peter",
+  family = "Desmet",
+  email = "peter.desmet@inbo.be",
+  comment = c(ORCID = "0000-0002-8442-8025")
+)
+```
+
+Create file paths:
+
+```{r}
+dir <- here::here("data", "processed", project_id, "dwc")
+record_id <- gsub("https://doi.org/10.5281/zenodo.", "", versioned_doi)
+zenodo_url <- paste0("https://zenodo.org/records/", record_id, "/export/json")
+datapackage_url <- paste0("https://zenodo.org/records/", record_id, "/files/datapackage.json")
+```
+
+## Create and write EML
+
+```{r}
+eml <- movepub::write_eml(
+  doi = versioned_doi,
+  directory = dir,
+  contact = contact,
+  derived_paragraph = TRUE
+)
+```
+
+Process EML:
+
+```{r}
+# Get description from Zenodo (with HTML, see https://github.com/inbo/movepub/issues/65)
+zenodo <- jsonlite::read_json(zenodo_url)
+description_full <- zenodo$metadata$description
+
+# ...
```

Comment:
@sannegovaert can you describe here what you are doing? Add more comments below if it is hard to explain in a single sentence.

Reply:
done!

Reply:
With `eml$dataset$abstract$para <- description` we are currently overwriting the derived paragraph. This should be fixed.

Reply:
fixed